src/cobra/flux_analysis/reaction.py

```diff
@@ -1,50 +1,52 @@
-# -*- coding: utf-8 -*-
-
-""" functions for analyzing / creating objective functions """
-
-from __future__ import absolute_import, division
+"""Provide functions for analyzing / creating objective functions."""
 
 from operator import attrgetter
+from typing import TYPE_CHECKING, Dict, Union
 from warnings import warn
 
-from cobra.core import Reaction
+from ..core import Reaction
+
+
+if TYPE_CHECKING:
+    from cobra import Model
 
 
-def assess(model, reaction, flux_coefficient_cutoff=0.001, solver=None):
-    """Assesses production capacity.
+def assess(
+    model: "Model",
+    reaction: Union[Reaction, str],
+    flux_coefficient_cutoff: float = 0.001,
+) -> Union[bool, Dict]:
+    """Assess production capacity.
 
-    Assesses the capacity of the model to produce the precursors for the
-    reaction and absorb the production of the reaction while the reaction is
-    operating at, or above, the specified cutoff.
+    Assess the capacity of the `model` to produce the precursors for the
+    `reaction` and absorb the production of the `reaction` while the
+    `reaction` is operating at, or above, the specified
+    `flux_coefficient_cutoff`.
 
     Parameters
     ----------
     model : cobra.Model
-        The cobra model to assess production capacity for
-
-    reaction : reaction identifier or cobra.Reaction
-        The reaction to assess
-
-    flux_coefficient_cutoff :  float
-        The minimum flux that reaction must carry to be considered active.
-
-    solver : basestring
-        Solver name. If None, the default solver will be used.
+        The cobra model to assess production capacity for.
+    reaction : str or cobra.Reaction
+        The reaction to assess.
+    flux_coefficient_cutoff : float, optional
+        The minimum flux that reaction must carry to be considered active
+        (default 0.001).
 
     Returns
     -------
     bool or dict
         True if the model can produce the precursors and absorb the products
-        for the reaction operating at, or above, flux_coefficient_cutoff.
-        Otherwise, a dictionary of {'precursor': Status, 'product': Status}.
-        Where Status is the results from assess_precursors and
-        assess_products, respectively.
+        for the reaction operating at, or above, `flux_coefficient_cutoff`.
+        Otherwise, a dictionary of {'precursor': Status, 'product': Status},
+        where 'Status' is the results from `assess_precursors` and
+        `assess_products`, respectively.
 
     """
     reaction = model.reactions.get_by_any(reaction)[0]
     with model as m:
         m.objective = reaction
-        if _optimize_or_value(m, solver=solver) >= flux_coefficient_cutoff:
+        if _optimize_or_value(m) >= flux_coefficient_cutoff:
```

Comment:
I would replace this with a direct call to the method.
```suggestion
        if m.slim_optimize(error_value=0.0) >= flux_coefficient_cutoff:
```

Reply:
I see that you have opted to remove the complete usage of `_optimize_or_value()` from the module. In that case, I will issue a deprecation warning for this as well.

Reply:
Since it was never intended for public use (name starts with `_`), you can remove it right away.

Reply:
That is true. I'm debating whether a private function should enjoy a deprecation notice or not. What do you think about that? 🤔